apps/dashboard/app/models/project.rb

```diff
@@ -27,33 +28,61 @@ def find(project_path)
     def dataroot
       Rails.logger.debug("project path is: #{OodAppkit.dataroot.join('projects')}")
 
-      OodAppkit.dataroot.join('projects').tap do |p|
-        p.mkpath unless p.exist?
+      OodAppkit.dataroot.join('projects').tap do |path|
+        path.mkpath unless path.exist?
       rescue StandardError => e
         Pathname.new('')
       end
     end
   end
 
+  validates :dir, presence: true
+  validates :dir, format: {
+
+    with: /\A[\w-]+\z/,
+    message: 'Directory may only contain letters, digits, dashes, and underscores'
+  }
+
   attr_reader :dir
+  delegate :icon, :name, :description, to: :manifest
+
+  def initialize(attributes = {})
+    @dir            = attributes.fetch(:dir, nil).to_s
+  end
 
-  def initialize(args = {})
-    @dir = args.fetch(:dir, nil).to_s
+  def save
+    result = update({ name: name, description: description, icon: icon })
```

Comment:
I couldn't get the `update(to_h)` or `update(:to_h)` to work. Passing the hash works, but I'm not sure if this is fragile in some way I'm missing.

Reply:
Maybe `self.to_h` to be explicit?

Reply:
Trying to use `self` always results in sometime of error of `No method to_h` being thrown. 

I ended up using just an empty hash, which looks to work, though I don't quite know why tbh.